ethdb/privateapi/ethbackend.go

```diff
@@ -270,6 +285,56 @@ func (s *EthBackendServer) EngineGetPayloadV1(ctx context.Context, req *remote.E
 	return nil, fmt.Errorf("unknown payload")
 }
 
+// EngineForkChoiceUpdatedV1, either states new block head or request the assembling of a new bloc
+func (s *EthBackendServer) EngineForkChoiceUpdatedV1(ctx context.Context, req *remote.EngineForkChoiceUpdatedRequest) (*remote.EngineForkChoiceUpdatedReply, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.config.TerminalTotalDifficulty == nil {
+		return nil, fmt.Errorf("not a proof-of-stake chain")
+	}
+	// Check if parent equate to the head
+	parent := gointerfaces.ConvertH256ToHash(req.Forkchoice.HeadBlockHash)
+	tx, err := s.db.BeginRo(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	headHeader, err := rawdb.ReadHeaderByHash(tx, parent)
+	if err != nil {
+		return nil, err
+	}
+
+	if atomic.LoadUint32(s.waitingForBeaconChain) == 0 || headHeader == nil {
```

Comment:
The previous check `parent != rawdb.ReadHeadBlockHash(tx)` is better since we might have the header in the DB but be on a different fork.

Reply:
we are reading the header by hash, which means it looks into blockhashes which does not do only canonical headers

Reply:
Yeah, that's my point. When our latest canonical header doesn't match headBlockHash, we should return "SYNCING" status and initiate a re-org.

Reply:
enrique will add it